### ISSUE TITLE
Bug: sort-keys is ignored in case object properties are used as a key

### ISSUE THREAD

**Issue by alex1701c** [eslint/eslint#18000]
### Environment

Reproducible in current playground

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>
Defaults of playground.
</details>

```js
/* eslint sort-keys: ["error", "asc"] */
/*eslint-env es6*/
const components = {
    // actions
    [MyComponent.identifier]: something,
    [AnotherComponent.identifier]: somethingElse,
}

const myComponents = {
    // actions
    [MyComponent]: something,
    [AnotherComponent]: somethingElse,
}
```

### What did you expect to happen?

This code was properly validated and I get "Expected object keys to be in ascending order. 'AnotherComponent' should be before 'MyComponent'.  (sort-keys)". I would have expected the same for the snippet above

### What actually happened?

There are no warnings. Not in the actual project or in the playgound with a minimal example.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHNvcnQta2V5czogW1wiZXJyb3JcIiwgXCJhc2NcIl0gKi9cbi8qZXNsaW50LWVudiBlczYqL1xuY29uc3QgY29tcG9uZW50cyA9IHtcbiAgICAvLyBhY3Rpb25zXG4gICAgW015Q29tcG9uZW50LmlkZW50aWZpZXJdOiBzb21ldGhpbmcsXG4gICAgW0Fub3RoZXJDb21wb25lbnQuaWRlbnRpZmllcl06IHNvbWV0aGluZ0Vsc2UsXG59Iiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Tanujkanti4441** [comment]
i think this issue is occurring because `sort-keys` rule is supposed to reports order of static property names or the identifier property key  but the example code 
```js
const components = {
    // actions
    [MyComponent.identifier]: something,
    [AnotherComponent.identifier]: somethingElse,
}
```
has member expression < then identifier as a key

**Comment by nzakas** [comment]
That's correct, `sort-keys` only sorts string values that are property keys. We do not sort computed key values.

We are no longer updating rules that enforce stylistic preferences, so won't be adding any option for this to the rule. However, you can always create your own custom rule to do that if you'd like.

**Comment by alex1701c** [comment]
>That's correct, sort-keys only sorts string values that are property keys. We do not sort computed key values.

Hmm okay, but the cut feels a bit off to me. If all dynamic property names were ignored, it would be perfectly reasonable.
